ddtrace/opentelemetry/span.go

```diff
@@ -142,26 +165,35 @@ func (s *span) SetStatus(code otelcodes.Code, description string) {
 // SetAttributes sets the key-value pairs as tags on the span.
 // Every value is propagated as an interface.
 func (s *span) SetAttributes(kv ...attribute.KeyValue) {
-	for _, attr := range kv {
-		s.SetTag(toSpecialAttributes(string(attr.Key), attr.Value))
+	for _, kv := range kv {
+		if k, v := toSpecialAttributes(string(kv.Key), kv.Value); k != "" {
+			s.attributes[k] = v
+		}
 	}
 }
 
 // toSpecialAttributes recognizes a set of span attributes that have a special meaning.
 // These tags should supersede other values.
 func toSpecialAttributes(k string, v attribute.Value) (string, interface{}) {
```

Comment:
nit: I renamed this to be called "reserved attributes" instead of "special attributes" in the RFC, just to make it a little clearer in the docs. We can either rename it in this PR, or a follow-up one later.